### PR TITLE
secboot, boot: TPM provisioning mode enum, introduce reprovisioning

### PR DESCRIPTION
**Pull request by bboozzoo**
Based on https://github.com/snapcore/snapd/pull/11828. Introduces an enum for TPM provisnioning modes and a flag that indicates a reprovisioning operation.

### CODE REVIEW

File: secboot/secboot.go [hunk 1]
```diff
@@ -87,6 +87,19 @@ type SealKeyModelParams struct {
 	KernelCmdlines []string
 }
 
+type TPMProvisionMode int
+
+const (
+	TPMProvisionNone TPMProvisionMode = iota
+	// TPMProvisionFull indicates a full provisioning of the TPM
+	TPMProvisionFull
+	// TPMPartialReprovision indicates a partial reprovisioning f the TPM
```
Comment (pedronis) [inline secboot/secboot.go:96]: typo: of

File: secboot/secboot_tpm.go [hunk 2]
```diff
@@ -338,14 +340,14 @@ func SealKeys(keys []SealKeyRequest, params *SealKeysParams) error {
 
 	authKey, err := sbSealKeyToTPMMultiple(tpm, sbKeys, &creationParams)
 	if err != nil {
+		logger.Noticef("seal key err: %v", err)
```
Comment (pedronis) [inline secboot/secboot_tpm.go:343]: s/err/error/

File: secboot/secboot_tpm.go [hunk 3]
```diff
@@ -282,7 +282,7 @@ func unlockEncryptedPartitionWithSealedKey(mapperName, sourceDevice, keyfile str
 
 // ProvisionTPM provisions the default TPM and saves the lockout authorization
 // key to the specified file.
-func ProvisionTPM(lockoutAuthFile string) error {
+func ProvisionTPM(mode TPMProvisionMode, lockoutAuthFile string) error {
```
Comment (pedronis) [inline secboot/secboot_tpm.go:285]: secboot_dummy.go needs to be changed too? it seems so from the warning

**Review by pedronis — APPROVED** [review]
thanks, small remarks

File: secboot/secboot_tpm.go [hunk 4]
```diff
@@ -338,14 +340,14 @@ func SealKeys(keys []SealKeyRequest, params *SealKeysParams) error {
 
 	authKey, err := sbSealKeyToTPMMultiple(tpm, sbKeys, &creationParams)
 	if err != nil {
+		logger.Debugf("seal key error: %v", err)
```
Comment (stolowski) [inline secboot/secboot_tpm.go:343]: Noticef maybe? Unless the returned err is shown anyway in the log?

**Review by stolowski — APPROVED** [review]
LGTM, thanks

File: secboot/secboot_tpm.go (hunk 4 above)
Comment (bboozzoo) [reply secboot/secboot_tpm.go:343]: Yes, the error bubbles up and is shown in the log.

Comment (stolowski) [reply secboot/secboot_tpm.go:343]: Do we need Debugf then?

**Pull request merged**